gemrb/core/Scriptable/Scriptable.cpp

```diff
@@ -1837,7 +1836,24 @@ bool Highlightable::TryUnlock(Actor *actor, bool removekey) const {
 
 	if (removekey) {
 		CREItem *item = NULL;
-		haskey->inventory.RemoveItem(KeyResRef,0,&item);
+		int result = haskey->inventory.RemoveItem(KeyResRef,0,&item);
+		if (result == -1) {
+			int i= haskey->inventory.GetSlotCount();
+			while (i--) {
+				//maybe we could speed this up if we mark bag items with a flags bit
+				const CREItem *itemslot = haskey->inventory.GetSlotItem(i);
+				if (!itemslot)
+					continue;
+				const Item *itemStore = gamedata->GetItem(itemslot->ItemResRef);
+				if (!itemStore)
+					continue;
+				if (core->CanUseItemType(SLOT_BAG, itemStore)) {
+					//the store is the same as the item's name
+					RemoveItemOfStore(itemslot->ItemResRef, KeyResRef);
```

Comment:
Is it important if it fails? I guess not.

Reply:
You could have multiple bags and it should look it in all of them.